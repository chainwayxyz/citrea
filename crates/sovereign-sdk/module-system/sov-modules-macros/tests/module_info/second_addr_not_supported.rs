use sov_modules_api::{Context, ModuleInfo, StateMap};

#[derive(ModuleInfo)]
struct TestStruct<C: Context> {
    #[address]
    address_1: C::Address,

    #[address]
    address_2: C::Address,

    #[state]
    test_state1: StateMap<u32, u32>,

    #[state]
    test_state2: StateMap<Vec<u8>, u64>,
}

fn main() {}
