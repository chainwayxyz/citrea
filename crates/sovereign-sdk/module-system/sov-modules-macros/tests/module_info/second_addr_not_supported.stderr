error: The `address` attribute is defined more than once, revisit field: address_2
 --> tests/module_info/second_addr_not_supported.rs:9:5
  |
9 |     address_2: C::Address,
  |     ^^^^^^^^^
